plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -163,6 +164,28 @@ const AddressForm = ( {
 					);
 				}
 
+				if ( field.type === 'select' ) {
+					if ( typeof field.options === 'undefined' ) {
+						return null;
+					}
```

Comment:
If we gracefully fail here, we should allow it to gracefully fail in registration.

Reply:
Graceful failures added in https://github.com/woocommerce/woocommerce/pull/42758/commits/25c12a5e7f81de53d8ee2eadbb1d7435fe23bad0